modules/formatter/tests/src/FormatterSpec.scala

```diff
@@ -0,0 +1,258 @@
+ *
+ *    https://disneystreaming.github.io/TOST-1.0.txt
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package smithytranslate
+package formatter
+
+import smithytranslate.formatter.parsers.IdlParser
+import smithytranslate.formatter.writers.IdlWriter.idlWriter
+import smithytranslate.formatter.writers.Writer._
+import munit.Location
+
+final class FormatterSpec extends munit.FunSuite {
+  private def formatTest(src: String, expected: String)(implicit
+      loc: Location
+  ) = {
+    IdlParser.idlParser
+      .parseAll(src)
+      .fold(
+        err => fail(s"Parsing failed with error: $err"),
+        idl => {
+          assertEquals(
+            idl.write,
+            expected
+          )
+        }
+      )
+  }
+
+  test("format test - keep new line after control section") {
+    val src = """|$version: "2.0"
+                 |
+                 |namespace test
+                 |
+                 |use alloy#uuidFormat
+                 |
+                 |long TargetingRuleId
+                 |""".stripMargin
+    val expected = """|$version: "2.0"
+                      |
+                      |namespace test
+                      |
+                      |use alloy#uuidFormat
+                      |
+                      |long TargetingRuleId
+                      |
+                      |""".stripMargin
+    formatTest(src, expected)
+  }
+
+  test("format test - handles triple comments") {
+    // this test is intended to catch instances where
+    // the line after a triple comment is moved onto the comment
+    // line. this example produces the following:
+    //  `/// Used to identify the disney platformstring PartnerName`
+    // this only happes if `long TargetingRuleId` is there
+    val src = """|$version: "2.0"
+                 |
+                 |namespace test
+                 |
+                 |use dslib#uuidFormat
+                 |
+                 |long TargetingRuleId
+                 |
+                 |/// Used to identify the disney platform
+                 |string PartnerName
+                 |
+                 |integer FeatureVersion
+                 |""".stripMargin
+    val expected = """|$version: "2.0"
+                      |
+                      |namespace test
+                      |
+                      |use dslib#uuidFormat
+                      |
+                      |long TargetingRuleId
+                      |
+                      |/// Used to identify the disney platform
+                      |string PartnerName
+                      |
+                      |integer FeatureVersion
+                      |
+                      |""".stripMargin
```

Comment:
there is always this newline that I hate and that'll get to in a few

Reply:
Do you want to do that in this PR or a different one?